boards/pygamer/Cargo.toml

```diff
@@ -29,18 +31,17 @@ version = "0.18.2"
 
 [dependencies.usb-device]
 optional = true
-version = "0.3.1"
+version = "0.3.2"
 
 [dev-dependencies]
-embedded-graphics = "0.7.1"
-embedded-hal-bus = "0.2.0"
+embedded-graphics = "0.8.1"
 embedded-sdmmc = "0.8.0"
 lis3dh = "0.4.3"
 micromath = "2.1"
-panic-halt = "0.2"
+panic-halt = "1"
 rtic = {version = "2.1.1", features = ["thumbv7-backend"]}
 smart-leds = "0.4"
-tinybmp = "0.3.1"
+tinybmp = "0.6"
```

Comment:
For the sake of completeness, could you add these features to Cargo.toml:

```toml
[features]
dma = ["atsamd-hal/dma"]
max-channels = ["dma", "atsamd-hal/max-channels"]
```

Reply:
I actually wondered whether these new features should be exposed in the BSPs, but I checked and noticed that there are other HAL features that are not exposed in Tier 1 BSPs, so I wasn't sure what determines whether a feature should be exposed or not. As a specific example, the HAL `defmt` feature is not exposed in the Tier 1 `metro_m4` BSP. Obviously, features are exposed if they are used in the BSP examples.

Adjacent question: Do we want to use DMA in any of the `pygamer` examples that use SPI? I noticed you did so for other Tier 1 BSPs. My thought was that for sure I would have the neopixels example use DMA SPI, but that is waiting on [this PR](https://github.com/smart-leds-rs/ws2812-spi-rs/pull/39) as we discussed @jbeaurivage.

Reply:
Reexporting the HAL features are essentially there for 3 reasons:

1. Obviously, if examples use the features, we need to reexport them
1. Even if no example uses them, having them reexported is convenient for the user; otherwise they'd have to specify `atsamd-hal` in their deps with the features they want, potentially leading to version mismatches
1. A more minor reason, but most of our CI scripts call `cargo clippy --examples --all-features` for each BSP. Reexporting the features helps us get a little more coverage in our test builds.

The fact that they aren't reexported in all T1 BSPs is just an oversight. I could always open a PR/issue eventually to get that sorted out.

Up to you to use DMA or not in the examples. I'm not opposed to having a little variety in our examples across BSPs. Again, more API surface being tested (if people do run the examples!), and it gives newcomers a glimpse into what is and isn't possible to do with the HAL.

There has been some discussion around standardizing the BSP examples, but that's certainly more than I'm willing to chew right now!

Reply:
With the latest commit, exposed these features for convenience. Only two current examples use SPI, so I'm going to hold off on demoing DMA SPI until the neopixel example is added pending the `ws2812-spi-rs` PR.

Not attempting to tackle the more general issue of exposing all HAL features in all the Tier 1 BSPs.